crates/turborepo-analytics/src/lib.rs

```diff
@@ -0,0 +1,397 @@
+                    source: CacheSource::Local,
+                    event: CacheEvent::Hit,
+                    hash: "".to_string(),
+                    duration: 0,
+                })
+                .unwrap();
+        }
+
+        expected_immediate_message(&mut rx).await;
+
+        let found = client.events();
+        assert_eq!(found.len(), 1);
+
+        let payloads = &found[0];
+        assert_eq!(payloads.len(), 10);
+
+        expect_timeout_then_message(&mut rx).await;
+        let found = client.events();
+        assert_eq!(found.len(), 2);
+
+        let payloads = &found[1];
+        assert_eq!(payloads.len(), 2);
+    }
+
+    #[tokio::test]
+    async fn test_closing() {
+        let (tx, mut rx) = mpsc::unbounded_channel();
+
+        let client = DummyClient {
+            events: Default::default(),
+            tx,
+        };
+
+        let (analytics_sender, analytics_handle) = start_analytics(
+            APIAuth {
+                token: "foo".to_string(),
+                team_id: "bar".to_string(),
+                team_slug: None,
+            },
+            client.clone(),
+        );
+
+        for _ in 0..2 {
+            analytics_sender
+                .send(AnalyticsEvent {
+                    session_id: None,
+                    source: CacheSource::Local,
+                    event: CacheEvent::Hit,
+                    hash: "".to_string(),
+                    duration: 0,
+                })
+                .unwrap();
+        }
+
+        let found = client.events();
+        assert!(found.is_empty());
+
+        analytics_handle.close().await.unwrap();
+        let found = client.events();
+        assert_eq!(found.len(), 1);
+        let payloads = &found[0];
+        assert_eq!(payloads.len(), 2);
+    }
+
+    #[test_case(
+      AnalyticsEvent {
+        session_id: Some("session-id".to_string()),
+        source: turborepo_vercel_api::CacheSource::Local,
+        event: turborepo_vercel_api::CacheEvent::Hit,
+        hash: "this-is-my-hash".to_string(),
+        duration: 58,
+      },
+      "with-id-local-hit"
+    )]
+    #[test_case(
+      AnalyticsEvent {
+        session_id: Some("session-id".to_string()),
+        source: turborepo_vercel_api::CacheSource::Remote,
+        event: turborepo_vercel_api::CacheEvent::Miss,
+        hash: "this-is-my-hash-2".to_string(),
+        duration: 21,
+      },
+      "with-id-remote-miss"
+    )]
+    #[test_case(
+        AnalyticsEvent {
+          session_id: None,
+          source: turborepo_vercel_api::CacheSource::Remote,
+          event: turborepo_vercel_api::CacheEvent::Miss,
+          hash: "this-is-my-hash-2".to_string(),
+          duration: 21,
+        },
+        "without-id-remote-miss"
+    )]
+    fn test_serialize_analytics_event(event: AnalyticsEvent, name: &str) {
+        let json = serde_json::to_string(&event).unwrap();
+        insta::assert_json_snapshot!(name, json);
+    }
```

Comment:
Nit: these tests should live in `turborepo-vercel-api` since that's where the serialization logic is defined.